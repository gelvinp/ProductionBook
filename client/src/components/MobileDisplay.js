import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Icon, Menu, Segment, Button } from 'semantic-ui-react'
import PDFViewer from '../containers/PDFViewerContainer.js'
import DocumentColumn from '../containers/DocumentColumnContainer.js'

class MobileDisplay extends Component {
  render() {
    const { fileOpen, deselectDocument, logOut } = this.props
    return fileOpen ? (
      <div
        style={{
          height: '100%',
          width: '100%',
          display: 'flex',
          flexDirection: 'column',
        }}
      >
        <Menu icon style={{ marginBottom: 0 }}>
          <Menu.Item onClick={deselectDocument}>
            <Icon name="chevron circle left" />
          </Menu.Item>
        </Menu>
        <PDFViewer mobile />
      </div>
    ) : (
      <div
        style={{
          height: '100%',
          width: '100%',
          display: 'flex',
          flexDirection: 'column',
        }}
      >
        <Segment
          inverted
          attached
          style={{
            marginBottom: 0,
            paddingTop: '0.25em',
            paddingBottom: '0.25em',
          }}
        >
          <div
            style={{
              display: 'flex',
              flexDirection: 'row',
              justifyContent: 'space-between',
              alignItems: 'center',
            }}
          >
            <p style={{ marginBottom: 0 }}>
              {process.env.REACT_APP_PROJECT} Production Book
            </p>
            <Button
              icon="close"
              floated="right"
              inverted
              basic
              size="mini"
              onClick={logOut}
            />
          </div>
        </Segment>
        <DocumentColumn mobile />
      </div>
    )
  }
}

MobileDisplay.propTypes = {
  fileOpen: PropTypes.bool.isRequired,
  deselectDocument: PropTypes.func.isRequired,
  logOut: PropTypes.func.isRequired,
}

export default MobileDisplay
