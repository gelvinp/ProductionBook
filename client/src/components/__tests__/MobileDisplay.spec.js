import React from 'react'
import MobileDisplay from '../MobileDisplay.js'
import { shallow } from 'enzyme'

describe('MobileDisplay', () => {
  it('Displays correctly', () => {
    const wrapper = shallow(
      <MobileDisplay
        fileOpen={false}
        deselectDocument={jest.fn()}
        logOut={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Displays a document column', () => {
    const wrapper = shallow(
      <MobileDisplay
        fileOpen={false}
        deselectDocument={jest.fn()}
        logOut={jest.fn()}
      />
    )
    expect(wrapper.find('Connect(DocumentColumn)').exists()).toBeTruthy()
    expect(wrapper.find('Connect(PDFViewer)').exists()).not.toBeTruthy()
  })

  it('Displays a PDF Viewer', () => {
    const wrapper = shallow(
      <MobileDisplay
        fileOpen={true}
        deselectDocument={jest.fn()}
        logOut={jest.fn()}
      />
    )
    expect(wrapper.find('Connect(PDFViewer)').exists()).toBeTruthy()
    expect(wrapper.find('Connect(DocumentColumn)').exists()).not.toBeTruthy()
  })

  it('Deselects the document', () => {
    const deselectDocument = jest.fn()
    const wrapper = shallow(
      <MobileDisplay
        fileOpen={true}
        deselectDocument={deselectDocument}
        logOut={jest.fn()}
      />
    )
    wrapper.find('MenuItem').simulate('click')
    expect(deselectDocument).toHaveBeenCalled()
  })

  it('Logs out', () => {
    const logOut = jest.fn()
    const wrapper = shallow(
      <MobileDisplay
        fileOpen={false}
        deselectDocument={jest.fn()}
        logOut={logOut}
      />
    )
    wrapper.find('Button').simulate('click')
    expect(logOut).toHaveBeenCalled()
  })
})
