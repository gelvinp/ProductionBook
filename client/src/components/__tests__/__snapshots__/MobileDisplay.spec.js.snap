// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileDisplay Displays correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Segment
    attached={true}
    inverted={true}
    style={
      Object {
        "marginBottom": 0,
        "paddingBottom": "0.25em",
        "paddingTop": "0.25em",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <p
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        Development
         Production Book
      </p>
      <Button
        as="button"
        basic={true}
        floated="right"
        icon="close"
        inverted={true}
        onClick={[MockFunction]}
        size="mini"
      />
    </div>
  </Segment>
  <Connect(DocumentColumn)
    mobile={true}
  />
</div>
`;
